File: 39주차/튜플.py
def solution(s):
    answer =[]
    arr= []
    s = s[2:-2]
    
    for a in s.split("},{"):
        tmp = list(map(int,a.split(",")))# 문자열을 각 배열로 만들어서 추가.
        arr.append(tmp)
    
    
    arr.sort(key=lambda x:len(x))# 배열 원소개수 작은것 부터 정렬.

    res = {}
    for r in arr:# 하나씩 값을 추가한다.
        for v in r:
            if v not in answer:
                res[v] = 1
    for key in res.keys():
        answer.append(key)
             
            
    return answer
# 프로그래머스 카카오인턴 문제

File: 47주차/튜플.js
function solution(s) {
  var answer = [];
  let arr = s.slice(2, s.length - 2).split("},{");

  arr = arr.map((v, i) => {
    return v.split(",");
  });
  let tmp = [];
  for (a of arr) {
    a = a.map((v, i) => {
      return parseInt(v);
    });
    tmp.push(a);
  }

  tmp.sort((a, b) => {
    // 배열 길이가 작은 순으로 정렬.
    return a.length - b.length;
  });

  let dict = new Set();
  for (let arr of tmp) {
    for (let v of arr) {
      if (!dict.has(v)) {
        // 없으면,
        dict.add(v);
      }
    }
  }
  answer = [...dict];
  return answer;
}
// 프로그래머스 - 카카오 튜플.
